lib/m3u8-parser.js

```diff
@@ -8,9 +8,14 @@ const util     = require('util');
  * @extends WritableStream
  * @constructor
  */
-var m3u8parser = module.exports = function() {
+var m3u8parser = module.exports = function(ytdlOptions) {
   var lastLine = '';
   var self = this;
+  self.ytdlSavedLines = [];
+  if (ytdlOptions) {
```

Comment:
Sets the YTDL options coming from the initial creation. We use these to determine if we need ytdl or not, and if it's the first time or not (see below).